src/plugins/vis_type_xy/public/components/detailed_tooltip.test.tsx

```diff
@@ -0,0 +1,237 @@
+      date: true,
+      intervalESUnit: 'h',
+      intervalESValue: 3,
+      interval: 10800000,
+      format: 'YYYY-MM-DD HH:mm',
+    },
+  },
+  y: [
+    {
+      accessor: 'col-2-1',
+      column: 2,
+      title: 'Count',
+      format: {
+        id: 'number',
+      },
+      aggType: 'count',
+      aggId: '1',
+      params: {},
+    },
+  ],
+  splitColumn: {
+    accessor: 'col-1-4',
+    column: 1,
+    title: 'Cancelled: Descending',
+    format: {
+      id: 'terms',
+      params: {
+        id: 'boolean',
+        otherBucketLabel: 'Other',
+        missingBucketLabel: 'Missing',
+      },
+    },
+    aggType: 'terms',
+    aggId: '4',
+    params: {},
+  },
+};
+
+const aspectsWithSplitRow = {
+  x: {
+    accessor: 'col-0-3',
+    column: 0,
+    title: 'timestamp per 3 hours',
+    format: {
+      id: 'date',
+      params: {
+        pattern: 'YYYY-MM-DD HH:mm',
+      },
+    },
+    aggType: 'date_histogram',
+    aggId: '3',
+    params: {
+      date: true,
+      intervalESUnit: 'h',
+      intervalESValue: 3,
+      interval: 10800000,
+      format: 'YYYY-MM-DD HH:mm',
+    },
+  },
+  y: [
+    {
+      accessor: 'col-2-1',
+      column: 2,
+      title: 'Count',
+      format: {
+        id: 'number',
+      },
+      aggType: 'count',
+      aggId: '1',
+      params: {},
+    },
+  ],
+  splitRow: {
+    accessor: 'col-1-4',
+    column: 1,
+    title: 'Cancelled: Descending',
+    format: {
+      id: 'terms',
+      params: {
+        id: 'boolean',
+        otherBucketLabel: 'Other',
+        missingBucketLabel: 'Missing',
+      },
+    },
+    aggType: 'terms',
+    aggId: '4',
+    params: {},
+  },
+};
+
+const header = {
+  seriesIdentifier: {
+    key:
+      'groupId{__pseudo_stacked_group-ValueAxis-1__}spec{area-col-1-1}yAccessor{col-1-1}splitAccessors{}smV{__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__}smH{__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__}',
+    specId: 'area-col-1-1',
+    yAccessor: 'col-1-1',
+    splitAccessors: {},
+    seriesKeys: ['col-1-1'],
+    smVerticalAccessorValue: '__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__',
+    smHorizontalAccessorValue: '__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__',
+  },
+  valueAccessor: 'y1',
+  label: 'Count',
+  value: 1611817200000,
+  formattedValue: '1611817200000',
+  markValue: null,
+  color: '#54b399',
+  isHighlighted: false,
+  isVisible: true,
+};
+
+const value = {
+  seriesIdentifier: {
+    key:
+      'groupId{__pseudo_stacked_group-ValueAxis-1__}spec{area-col-1-1}yAccessor{col-1-1}splitAccessors{}smV{__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__}smH{__ECH_DEFAULT_SINGLE_PANEL_SM_VALUE__}',
+    specId: 'area-col-1-1',
+    yAccessor: 'col-1-1',
+    splitAccessors: [],
+    seriesKeys: ['col-1-1'],
+    smVerticalAccessorValue: 'true',
+    smHorizontalAccessorValue: 'false',
+  },
+  valueAccessor: 'y1',
+  label: 'Count',
+  value: 52,
+  formattedValue: '52',
+  markValue: null,
+  color: '#54b399',
+  isHighlighted: true,
+  isVisible: true,
+};
```

Comment:
I would advise putting these verbose mock objects into a `detailed_tooltip.mock.ts` file or some more general name if used outside of just these tests. This would clean this up a bit and focus on just the test cases in this file.

Reply:
I agree with both comments Nick. You can see the fixes here 67afa7b909d163ed05bec011f863cc3a3a666722